src/ray/raylet/scheduling_resources.cc

```diff
@@ -302,6 +375,53 @@ std::string ResourceIds::ToString() const {
   return return_string;
 }
 
+void ResourceIds::UpdateCapacity(int64_t new_capacity) {
+  // Assert the new capacity is positive for sanity
+  RAY_CHECK(new_capacity >= 0);
+  double capacity_delta = new_capacity - total_capacity_;
+  if (capacity_delta < 0) {
+    DecreaseCapacity(-1 * capacity_delta);
+  } else {
+    IncreaseCapacity(capacity_delta);
+  }
+}
+
+void ResourceIds::IncreaseCapacity(int64_t increment_quantity) {
+  // Adjust with decrement_backlog_
+  double actual_increment_quantity = std::max(0.0, increment_quantity-decrement_backlog_);
```

Comment:
Why are we turning an `int` into a `double`?

Reply:
oh sorry I missed this while refactoring - fixed now